test/test_models_detection_utils.py

```diff
@@ -58,6 +58,34 @@ def test_transform_copy_targets(self):
         self.assertTrue(torch.equal(targets[0]['boxes'], targets_copy[0]['boxes']))
         self.assertTrue(torch.equal(targets[1]['boxes'], targets_copy[1]['boxes']))
 
+    def test_normalize_integer(self):
+        transform = GeneralizedRCNNTransform(
+            300, 500,
+            torch.randint(0, 255, (3,)),
+            torch.randint(0, 255, (3,))
+        )
+        image = [torch.randint(0, 255, (3, 200, 300), dtype=torch.uint8)]
+        targets = [{'boxes': torch.randint(0, 255, (3, 4))}]
+        image_list, _ = transform(image, targets)  # noqa: F841
+        # check that original images still have uint8 dtype
+        for img in image:
+            self.assertTrue(img.dtype == torch.uint8)
+        # check that the resulting images have float32 dtype
+        self.assertTrue(image_list.tensors.dtype == torch.float32)
+        # check that no NaN values are produced
+        self.assertFalse(torch.any(torch.isnan(image_list.tensors)))
```

Comment:
@Wadaboa I'm still unable to see this test failing on master without your mitigation. We need to reproduce the problem, so that we are certain that we provide the right fix. Let me know your thoughts.

Reply:
I just went through the code that initially produced the issue for me and I realized the following: the mentioned error can only happen when the input image has `uint8` dtype (so, in [0, 255] range) and the mean/std parameters are instead in [0, 1]. In this way, the output given by normalization will be all `inf` (not NaN as shown in the tests). 

But in this case image and mean/std would have different ranges. So, is it something that should still be checked? Or maybe I reported an error which should alert the user to check its input and this PR is not needed?

Reply:
Thanks for investigating. I agree, there are too many corner-cases to cover for uint8 inputs. This specific piece of code generally expects the input image to be floating type, so if it's not it might be worth throwing an error. 

I think what we should do is close this PR to keep things clean and open a new one where you put a type check on top of normalize and throw an exception if the input is not floating point. What you think?

Reply:
I agree. I'll close this PR and open a new one for the type check. Thank you.